---
title: "README"
output: 
    md_document:
    variant: github_markdown
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(message = FALSE, comment = '', prompt = TRUE)
```

#### Install

First, we need to install `devtools`, if not already installed.

```{r eval=FALSE}
install.packages("devtools")
```

Then, we install `hlahaps` using the `install_github()` function.
```{r eval=FALSE}
devtools::install_github("VitorAguiar/hlahaps")
```

Now, we can load `hlahaps` as any other R package using the base function `library()`.

```{r eval=FALSE}
library(hlahaps)
```

```{r echo=FALSE}
devtools::load_all("~/hlahaps")
```

#### Usage

When we load the package, the [Gourraud et al. (2014) data](http://dx.doi.org/10.1371/journal.pone.0097282) is loaded. 

```{r}
pag
```

We will use the individual HG00096 to demonstrate how the `query_nmdp()` function works:

```{r}
test_ind <- dplyr::filter(pag, subject == "HG00096")
test_ind
```

Applying `query_nmdp()`:

```{r}
query_nmdp(test_ind)
```

`query_nmdp()` returns a list with 4 elements:

1. A data.frame with the original haplotypes
2. A data.frame with the haplotypes found at the `nmdp` table
3. A data.frame with the haplotypes which were not found at the `nmdp` table
4. A data.frame with possible haplotypes at `nmdp` table given the individual's alleles

It is possible to apply `query_nmdp()` to the whole data, e.g. by using `plyr::dlply()` with a parallel backend provided by `doMC::registerDoMC()`:

```{r}
# using 36 cores:
doMC::registerDoMC(36)

results_list <- plyr::dlply(pag, ~subject, . %>% query_nmdp(), .parallel = TRUE)
```

With a little hack using `purrr::transpose()` and `plyr::ldply()` it is possible to compile a data.frame with info for all individuals. For example, let's create a data.frame with all the haplotypes found in the NMDP table:

```{r}
haps_found <-
  purrr::transpose(results_list)$`haplotypes found` %>%
  plyr::ldply(rbind, .id = "subject") %>%
  tibble::as_tibble()

haps_found
```
