---
title: "README"
output: 
    md_document:
    variant: github_markdown
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(message=FALSE, comment='', prompt = TRUE)
```

When we attach the package, the [Gourraud et al. (2014) data](http://dx.doi.org/10.1371/journal.pone.0097282) is loaded. We will use the individual HG00096 to illustrate how the `get_hla_haps()` function works:

```{r eval=FALSE}
library(hlahaps)
```

```{r echo=FALSE}
devtools::load_all("~/hlahaps")
```

```{r}
pag

test_ind <- subset(pag, subject == "HG00096")
test_ind

get_hla_haps(test_ind)
```

`get_hla_haps` returns a list with 4 elements:

1. A data.frame with the original haplotypes
2. A data.frame with the haplotypes found at the `nmdp` table
3. A data.frame with the haplotypes which were not found at the `nmdp` table
4. A data.frame with possible haplotypes at `nmdp` table given the individual's alleles

It is possible to apply `get_hla_haps()` to the whole data, e.g. by using `plyr::dlply()` with a parallel backend provided by `doMC::registerDoMC()`:

```{r eval=FALSE}
n_cores <- 50
doMC::registerDoMC(n_cores)

results_list <- plyr::dlply(pag, ~subject, . %>% get_hla_haps, .parallel = TRUE)
```
